src/main/java/com/nike/cerberus/client/HttpClientFactory.java

```diff
@@ -0,0 +1,192 @@
+     */
+    public OkHttpClient getGenericClient() {
+        return new OkHttpClient.Builder()
+                .hostnameVerifier(new NoopHostnameVerifier())
+                .connectTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                .writeTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                .readTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                .build();
+    }
+
+    /**
+     * Downloads the CA Chains from S3 for the Cerberus certs and creates a client that can talk to the
+     * individual CMS instances without going through the alb, or manually having to add the chain to the trust store.
+     */
+    public OkHttpClient getGenericClientWithCustomTruststore() {
+        try {
+            List<String> caChains = new LinkedList<>();
+            for (CertificateInformation certInfo : configStore.getCertificationInformationList()) {
+                String certificateName = certInfo.getCertificateName();
+                caChains.add(configStore.getCertPart(certificateName, ConfigConstants.CERT_PART_CA)
+                        .orElseThrow(() -> new RuntimeException("Failed to download ca chain")));
+            }
+
+            Buffer buffer = new Buffer();
+            caChains.forEach(buffer::writeUtf8);
+
+            X509TrustManager trustManager = trustManagerForCertificates(buffer.inputStream());
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(null, new TrustManager[] { trustManager }, null);
+            SSLSocketFactory sslSocketFactory = sslContext.getSocketFactory();
+
+            return new OkHttpClient.Builder()
+                    .hostnameVerifier(new NoopHostnameVerifier())
+                    .sslSocketFactory(sslSocketFactory, trustManager)
+                    .connectTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                    .writeTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                    .readTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
+                    .build();
+        } catch (GeneralSecurityException e) {
+            throw new RuntimeException("Failed to create ok http client with custom trust manager from ca certs downloaded from S3");
+        }
+    }
+
+    /**
+     * https://github.com/square/okhttp/blob/master/samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java
+     *
+     * Returns a trust manager that trusts {@code certificates} and none other. HTTPS services whose
+     * certificates have not been signed by these certificates will fail with a {@code
+     * SSLHandshakeException}.
+     *
+     * <p>This can be used to replace the host platform's built-in trusted certificates with a custom
+     * set. This is useful in development where certificate authority-trusted certificates aren't
+     * available. Or in production, to avoid reliance on third-party certificate authorities.
+     *
+     *
+     * <h3>Warning: Customizing Trusted Certificates is Dangerous!</h3>
```

Comment:
I'm not sure I understand this warning and the paragraph below all of the way.  The language seems alarmist and it might be better to outline what you consider "safe" vs. "dangerous".

Reply:
That was copy pasted from the cook book recipe from OkHttp we can tone it down

Reply:
- [ ] tone down language